cmake/modules/Distutils.cmake

```diff
@@ -32,15 +32,27 @@ endfunction(distutils_install_module)
 function(distutils_add_cython_module name src)
   get_property(compiler_launcher GLOBAL PROPERTY RULE_LAUNCH_COMPILE)
   get_property(link_launcher GLOBAL PROPERTY RULE_LAUNCH_LINK)
-  set(PY_CC \"${compiler_launcher} ${CMAKE_C_COMPILER}\")
-  set(PY_CXX \"${compiler_launcher} ${CMAKE_CXX_COMPILER}\")
-  set(PY_LDSHARED \"${link_launcher} ${CMAKE_C_COMPILER} -shared\")
+  # When using ccache CMAKE_C_COMPILER is ccache executable absolute path
+  # and the actual C compiler is CMAKE_C_COMPILER_ARG1.
+  # However with a naive
+  # set(PY_CC ${compiler_launcher} ${CMAKE_C_COMPILER} ${CMAKE_C_COMPILER_ARG1})
+  # distutils tries to execve something like "/usr/bin/cmake gcc" and fails.
+  # Removing the leading whitespace from CMAKE_C_COMPILER_ARG1 helps to avoid
+  # the failure.
+  string(REGEX REPLACE "^[ ]+(.+)[ ]*$" \\1 _actual_c_compiler "${CMAKE_C_COMPILER_ARG1}")
```

Comment:
nit, the scopes of `_actual_c_compiler` and `_actual_cxx_compiler` are limited in the function, so no need to prefix them with `_`. naming the local variables using the lower case chars would suffice. also, i think it might be easier to understand if you could
```cmake
string(STRIP "${CMAKE_C_COMPILER_ARG1}" c_compiler_arg1)
```
and set `PY_CC`  like
```cmake
set(PY_CC ${compiler_launcher} ${CMAKE_C_COMPILER} ${c_compiler_arg1})
```

i can hardly get my head around `${CMAKE_C_COMPILER} ${_actual_c_compiler}` even with your comments. but it would be easier to digest if one just wants to strip off the spaces around the compiler flags before passing it to the compiler.